src/why3prop.ml

```diff
@@ -0,0 +1,127 @@
+(** Translation of lambdapi goals to why3 goals in propositional case *)
+
+open Terms
+open Extra
+
+exception NoGoalTranslation
+
+(* a map from lambdapi terms to Why3 constants *)
+type cnst_table = (term * Why3.Term.lsymbol) list
+
+(* a map from labels to Why3 terms. *)
+type ctxt_labels = (string * Why3.Term.term) list
+
+(* number of axioms proved whith the Why3 tactic *)
+let nbr_axioms : int ref = ref 0
+
+(** [get_newname ()] generates a new axiom name. *)
+let get_newname () : string =
```

Comment:
It's safer to let nbr_axioms be a variable local to get_newname so that nobody else can change its value:
```
let get_name : unit -> string =
  let nbr_axioms = ref 0 in fun () ->
  ...
```

Reply:
If it's a local variable, the next time a user calls the tactic `why3` the function `get_name` is called, and we will always have `"Why3axiom_0"` as a name !

Reply:
No we won't if you write it as Frederic did, because the reference is created before the function is defined. (Note the ```fun () -> ```.)

Reply:
Ah, I see. I didn't know that when a function is defined with a local environment, it will use this environment in each call even if the environment changes.
Fixed in 1b0b18e2a2198f6d225dc6a0d6ea5d721e425441